pkg/kubelet/kubelet_pods_test.go

```diff
@@ -3781,6 +3781,101 @@ func Test_generateAPIPodStatus(t *testing.T) {
 	}
 }
 
+func Test_generateAPIPodStatusForInPlaceVPAEnabled(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.InPlacePodVerticalScaling, true)()
+	testContainerName := "ctr0"
+	testContainerID := kubecontainer.ContainerID{Type: "test", ID: testContainerName}
+
+	CPU1AndMem1G := v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("1Gi")}
+	CPU1AndMem1GAndStorage2G := CPU1AndMem1G.DeepCopy()
+	CPU1AndMem1GAndStorage2G[v1.ResourceEphemeralStorage] = resource.MustParse("2Gi")
+
+	CPU1AndMem1GAndCustomAndStorage2GSpec := v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("1Gi"), "eni-ip": resource.MustParse("1")}
```

Comment:
where is the `eni-ip` from? is it from our internal system? Let's use something like `unknown-resource` instead of specific names